pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -80,19 +84,93 @@ func ValidateKubeSchedulerConfiguration(cc *config.KubeSchedulerConfiguration) f
 	return allErrs
 }
 
-func validateKubeSchedulerProfile(path *field.Path, profile *config.KubeSchedulerProfile) field.ErrorList {
-	allErrs := field.ErrorList{}
+func validateKubeSchedulerProfile(path *field.Path, profile *config.KubeSchedulerProfile) []error {
+	var errs []error
 	if len(profile.SchedulerName) == 0 {
-		allErrs = append(allErrs, field.Required(path.Child("schedulerName"), ""))
+		errs = append(errs, field.Required(path.Child("schedulerName"), ""))
 	}
-	return allErrs
+	errs = append(errs, validatePluginConfig(path, profile)...)
+	return errs
+}
+
+func fromAggregate(agg utilerrors.Aggregate) field.ErrorList {
+	if agg == nil {
+		return nil
+	}
+	errs := agg.Errors()
+	list := make(field.ErrorList, len(errs))
+	for i := range errs {
+		list[i] = errs[i].(*field.Error)
+	}
+	return list
+}
+
+type argsToValiateFunc struct {
+	args         interface{}
+	validateFunc interface{}
+}
+
+func validatePluginConfig(path *field.Path, profile *config.KubeSchedulerProfile) []error {
+	var errs []error
+	m := map[string]argsToValiateFunc{
+		"DefaultPreemption":           {&config.DefaultPreemptionArgs{}, ValidateDefaultPreemptionArgs},
+		"InterPodAffinity":            {&config.InterPodAffinityArgs{}, ValidateInterPodAffinityArgs},
+		"NodeAffinity":                {&config.NodeAffinityArgs{}, ValidateNodeAffinityArgs},
+		"NodeLabel":                   {&config.NodeLabelArgs{}, ValidateNodeLabelArgs},
+		"NodeResourcesFitArgs":        {&config.NodeResourcesFitArgs{}, ValidateNodeResourcesFitArgs},
+		"NodeResourcesLeastAllocated": {&config.NodeResourcesLeastAllocatedArgs{}, ValidateNodeResourcesLeastAllocatedArgs},
+		"NodeResourcesMostAllocated":  {&config.NodeResourcesMostAllocatedArgs{}, ValidateNodeResourcesMostAllocatedArgs},
+		"PodTopologySpread":           {&config.PodTopologySpreadArgs{}, ValidatePodTopologySpreadArgs},
+		"RequestedToCapacityRatio":    {&config.RequestedToCapacityRatioArgs{}, ValidateRequestedToCapacityRatioArgs},
+		"VolumeBinding":               {&config.VolumeBindingArgs{}, ValidateVolumeBindingArgs},
+	}
+
+	seenPluginConfig := make(sets.String)
+	for i := range profile.PluginConfig {
+		pluginConfigPath := path.Child("pluginConfig").Index(i)
+		name := profile.PluginConfig[i].Name
+		args := profile.PluginConfig[i].Args
+		if seenPluginConfig.Has(name) {
+			errs = append(errs, field.Duplicate(pluginConfigPath, name))
+		} else {
+			seenPluginConfig.Insert(name)
+		}
+		if plugin, ok := m[name]; ok {
+			// type mismatch, no need to validate the `args`.
+			if reflect.TypeOf(args) != reflect.TypeOf(plugin.args) {
```

Comment:
you can do `plugin.validateFunc.In(0)` to get the type of the parameter. And then you don't need `argsToValiateFunc`, just the method

Reply:
Done